memoize.js

```diff
@@ -0,0 +1,31 @@
+let cache = {};
+
+function memoizedById(id) {
+  return memoizedQueryBy('#' + id);
+}
+
+function memoizedQueryBy(selector) {
+  if (selector in cache) {
+    console.log('Fetching from cache');
+    return cache[selector];
+  }
+  console.log('Querying Element');
+  let result = document.querySelector(selector);
+  cache[selector] = result;
+  return result;
+}
+
+const findById = memoizedById;
```

Comment:
I believe the exercise was to offer replacements for the `getElementById` and `querySelectorAll` methods that are globally available. The name should be consistent.